src/containers/shared/components/Streams.jsx

```diff
@@ -330,6 +330,12 @@ class Streams extends Component {
     })
   }
 
+  async setStateAsync(state) {
```

Comment:
Is there a reason to not use other lifecycle methods to manage this? If the component unmounts you could end up with react errors

Reply:
What do you mean by other lifecycle methods? I didn't really want to get into a full refactor right now.
Honestly, this file is already sort of a mess (and probably has a ton of little bugs) and @ckniffen has a WIP refactor of all of this to use TypeScript/hooks.

Reply:
Why didn't the functional version of setState in the existing code not already work?

Reply:
It did, but it resulted in errors in local tests. I'm not really sure why it worked in CI but not locally, but I googled around and this seemed to fix it.

Reply:
Yes.
https://github.com/ripple/explorer/blob/staging/src/containers/shared/components/Streams.jsx#L401-L415
Lines 414-415 would execute before the state was finished updating. According to some StackOverflow research, this is the best way to resolve this.

Reply:
Ok makes sense, thanks